### PR TITLE
add sqlite3 storage

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

**Comment by fiatjaf** [comment]
Thank you!

Am I right in assuming there is no indexing at all for the tag queries?

**Comment by fiatjaf** [comment]
Would the best approach for having efficient tag queries to create a new table with the tags?

**Comment by mattn** [comment]
This is workaround for sqlite3 that stored procedure like postgresql does not work on sqlite3. I'm okay that I'll fix that same if you fix postgresql so that. :+1:

File: storage/sqlite3/query.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		params = append(params, filter.Since.Unix())
+	}
+	if filter.Until != nil {
+		conditions = append(conditions, "created_at < ?")
+		params = append(params, filter.Until.Unix())
+	}
+	if filter.Search != "" {
+		conditions = append(conditions, "content LIKE ?")
+		s := filter.Search
+		if ss, err := strconv.Unquote(s); err == nil {
```
Comment (mattn) [inline storage/sqlite3/query.go:128]: This Unquote is workaround that go-nostr always quote for Filter.Search.

https://github.com/nbd-wtf/go-nostr/pull/68

Comment (mattn) [reply storage/sqlite3/query.go:128]: https://github.com/nbd-wtf/go-nostr/pull/68 was merged just now. I'll update this PR.

**Comment by fiatjaf** [comment]
> This is workaround for sqlite3 that stored procedure like postgresql does not work on sqlite3. I'm okay that I'll fix that same if you fix postgresql so that. ￼

I don't understand what you mean here. I was just asking what was the best way since you probably know SQLite much better than I do. No need to implement that for this PR to be merged. I'll merge it as soon as you update the workaround.

**Comment by mattn** [comment]
updated go.mod

**Comment by mattn** [comment]
Please wait to remove Unquote

**Comment by mattn** [comment]
Fixed.

**Pull request merged**